docs/content/how-to/azure/create-azure-cluster-with-options.md

```diff
@@ -23,6 +23,81 @@ There are a few prerequisites for encrypting the OS disks on the Azure VMs:
+FILE_SP_APP_ID=$(az ad sp create-for-rbac --name "${AZURE_FILE_SP_NAME}" --create-cert --cert "${AZURE_FILE_SP_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+NODEPOOL_MGMT_APP_ID=$(az ad sp create-for-rbac --name "${NODEPOOL_MGMT}" --create-cert --cert "${NODEPOOL_MGMT}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+CLOUD_PROVIDER_APP_ID=$(az ad sp create-for-rbac --name "${CLOUD_PROVIDER_SP_NAME}" --create-cert --cert "${CLOUD_PROVIDER_SP_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+CNCC_APP_ID=$(az ad sp create-for-rbac --name "${CNCC_NAME}" --create-cert --cert "${CNCC_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+CONTROL_PLANE_APP_ID=$(az ad sp create-for-rbac --name "${CONTROL_PLANE_SP_NAME}" --create-cert --cert "${CONTROL_PLANE_SP_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+IMAGE_REGISTRY_APP_ID=$(az ad sp create-for-rbac --name "${IMAGE_REGISTRY_SP_NAME}" --create-cert --cert "${IMAGE_REGISTRY_SP_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+INGRESS_APP_ID=$(az ad sp create-for-rbac --name "${INGRESS_SP_NAME}" --create-cert --cert "${INGRESS_SP_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+```
+
+### Save service principal IDs and certificate names to a JSON file
+```
+SP_FILE="service-principals.json"
+
+# Save service principal IDs and certificate names to a JSON file
+OUTPUT_FILE="service-principals.json"
+
+cat <<EOF > SP_FILE
+{
+    "cloudProvider": {
+        "certificateName": "${CLOUD_PROVIDER_SP_NAME}",
+        "clientID": "${CLOUD_PROVIDER_APP_ID}"
+    },
+    "controlPlaneOperator": {
+        "certificateName": "${CONTROL_PLANE_SP_NAME}",
+        "clientID": "${CONTROL_PLANE_APP_ID}"
+    },
+    "disk": {
+        "certificateName": "${AZURE_DISK_SP_NAME}",
+        "clientID": "${DISK_SP_APP_ID}"
+    },
+    "file": {
+        "certificateName": "${AZURE_FILE_SP_NAME}",
+        "clientID": "${FILE_SP_APP_ID}"
+    },
+    "imageRegistry": {
+        "certificateName": "${IMAGE_REGISTRY_SP_NAME}",
+        "clientID": "${IMAGE_REGISTRY_APP_ID}"
+    },
+    "ingress": {
+        "certificateName": "${INGRESS_SP_NAME}",
+        "clientID": "${INGRESS_APP_ID}"
+    },
+    "network": {
+        "certificateName": "${CNCC_NAME}",
+        "clientID": "${CNCC_APP_ID}"
+    },
+    "nodePoolManagement": {
+        "certificateName": "${NODEPOOL_MGMT}",
+        "clientID": "${NODEPOOL_MGMT_APP_ID}"
+    },
+    "managedIdentitiesKeyVault": {
+        "name": "${KV_NAME}",
+        "tenantID": "$(az account show --query tenantId -o tsv)"
```

Comment:
Maybe this should just be predefined above. Maybe the MI KV can be in a different tenant than the use running this.